src/lib/Microsoft.Health.Fhir.Ingest.Template/HandlerProxyTemplateFactory.cs

```diff
@@ -12,8 +14,29 @@ public abstract class HandlerProxyTemplateFactory<TInput, TOutput> :
         IResponsibilityHandler<TInput, TOutput>
         where TOutput : class
     {
+        private readonly IList<string> _serializationErrors = new List<string>();
+
+        protected IList<string> SerializationErrors
+        {
+            get { return _serializationErrors; }
+        }
+
         public abstract TOutput Create(TInput jsonTemplate);
 
         TOutput IResponsibilityHandler<TInput, TOutput>.Evaluate(TInput request) => Create(request);
+
+        protected JsonSerializer GetJsonSerializer()
+        {
+            _serializationErrors.Clear();
+            return JsonSerializer.Create(new JsonSerializerSettings
+            {
+                MissingMemberHandling = MissingMemberHandling.Error,
+                Error = (sender, args) =>
+                {
+                    _serializationErrors.Add(args.ErrorContext.Error.Message);

```

Comment:
Handling the error like doesn't seem like the best.  Have you tried catching serialization errors from the caller instead of internalizing this?  My fear is we will have a complete but no functional template as part of the collection.  It would be better to let this template error, catch and record then continue with others.

Reply:
The Newtonsoft APIs give the option to validate JToken against an schema predefined https://www.newtonsoft.com/json/help/html/JTokenIsValid.htm. Or, it can use given serializer with customized error handler registered to stop it from throwing exception directly, in which we can get a full list of errors.  

This is not logically straightforward since the invocation happened to run two logics together: deserialization and validation. That's against the single responsibility.

In next revision, if we're going to homemade this by us, I'd like to create a framework that has the validation interface to retrieve the data schema requirements and other business validation requirements and run them on top of a de-serialized POCO.